### PR TITLE
CVE-2017-5648

### PR DESCRIPTION
**Pull request by naia4897**
Data curation project looking through investigation of bug with Apache Tomcat application listeners.

### CODE REVIEW

File: cves/CVE-2017-5648.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
```
Comment (griffinseibold) [inline cves/CVE-2017-5648.yml:10]: You have six other people's CVE files in this pull request which I'm not sure is suppose to happen.

File: cves/CVE-2017-5648.yml [hunk 2]
```diff
@@ -43,12 +43,15 @@ description_instructions: |
...
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
-  While investigating bug 60718, it was noticed that some calls application
-  listeners did not use the appropriate facade object. when running an untrusted
-  application under a security manager, it was possible for that application to
-  retain a references to the request or response objects and thereby acces or
-  modify information associated with another web application.
+description: |
+  While investigating bug 60718, it was noticed that some calls to application
```
Comment (griffinseibold) [inline cves/CVE-2017-5648.yml]: I would recommend getting less technical in this description, version numbers are not needed and some of the language is very technical.

File: cves/CVE-2017-5648.yml [hunk 3]
```diff
@@ -43,12 +43,15 @@ description_instructions: |
...
+  listeners in Apache Tomcat 9.0.0.M1 to 9.0.0.M17, 8.5.0 to 8.5.11, 8.0.0.RC1
+  to 8.0.41, and 7.0.0 to 7.0.75 did not use the appropriate facade object.
+  When running an untrusted application under a SecurityManager, it was therefore
+  possible for that untrusted application to retain a reference to the request or
+  response object and thereby access and/or modify information associated with
+  another web application. This issue came about due to requestDestroyed not
+  always being called after client disconnects.
```
Comment (griffinseibold) [inline cves/CVE-2017-5648.yml]: I see you mention what the fix was under the fixes section, could be worth adding a sentence here to explain how this vulnerability was resolved.

**Comment by griffinseibold** [comment]
Good job explaining what the vulnerability was and providing great detail. Very interesting how one of the people who responded tried to blame it on another bug, I feel as though this is a result of developers thinking that they know their system the best.

Upvotes: 3

File: cves/CVE-2017-5648.yml [hunk 4]
```diff
@@ -43,7 +37,14 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  UPDATE While investigating bug 60718, it was noticed that some calls to application
```
Comment (jmrossi98) [inline cves/CVE-2017-5648.yml]: The beginning part seems similar to the CVE website description. Maybe switch things up in your own words, specifically in this part.

File: cves/CVE-2017-5648.yml [hunk 5]
```diff
@@ -208,4 +228,15 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    It appears the mistakes stem from requirements not being verified. The differing
+    interpretation of how to handle the request in or out of scope is vague to the
```
Comment (jmrossi98) [inline cves/CVE-2017-5648.yml]: Add an 's' to interpretation

**Review by jmrossi98 — COMMENTED** [review]
Great explanation. Maybe try to refine the description.
Upvotes: 3

File: cves/CVE-2017-5648.yml (hunk 1 above)
Comment (naia4897) [reply cves/CVE-2017-5648.yml:10]: Than you for bringing this to my attention. Sorry for the confusion.

File: cves/CVE-2017-5648.yml (hunk 5 above)
Comment (naia4897) [reply cves/CVE-2017-5648.yml]: Thank you!

**Pull request closed without merging**